lib/data/source/local/local_datasource.dart

```diff
@@ -35,6 +35,16 @@ class LocalDataSource {
     await pokemonBox.putAll(pokemonsMap);
   }
 
+  Future<List<PokemonHiveModel>> searchPokemon({String pokemonName}) async {
+    final pokemonBox = Hive.box<PokemonHiveModel>(PokemonHiveModel.boxKey);
+
+    final pokemons = List.generate(pokemonBox.length, (index) => pokemonBox.getAt(index));
+
+    final filteredPokemons = pokemons.where((pokemon) => pokemon.name.contains(pokemonName)).toList();
```

Comment:
use `toLowerCase()` for both `pokemonName` and `pokemon.name` to allow user to search by both uppercase and lowercase